client/app/src/main/java/io/mosip/registration/app/activites/PacketListActivity.java

```diff
@@ -0,0 +1,302 @@
+
+        //to display back button
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setTitle(R.string.registrations_title);
+        getSupportActionBar().setSubtitle(R.string.registrations_subtitle);
+        init();
+        swipeRefreshLayout = findViewById(R.id.packet_list_swipe_layout);
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                swipeRefreshLayout.setRefreshing(true);
+                registrationPacketViewModel.refreshPacketStatus();
+                swipeRefreshLayout.setRefreshing(false);
+                adapter.notifyDataSetChanged();
+            }
+        });
+        auditManagerService.audit(AuditEvent.LOADED_REG_LISTING, Components.REG_PACKET_LIST);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        uploadOverWifiOnly = sharedPreferences.getBoolean("wifi_only", false);
+    }
+
+    private void init() {
+        actionCallback = new ActionCallback();
+        registrationPacketViewModel = new RegistrationPacketViewModel(packetService, dateUtil);
+        list = registrationPacketViewModel.getList();
+        adapter = new RegistrationPacketListAdapter(this, list);
+
+        bi.packetList.setLayoutManager(new LinearLayoutManager(this));
+        bi.packetList.setHasFixedSize(true);
+        bi.packetList.setAdapter(adapter);
+        adapter.setItemClick(new RegistrationPacketListAdapter.OnItemClick() {
+            @Override
+            public void onItemClick(View view, RegistrationPacketModel registrationPacketModel, int position) {
+                toggleActionBar(position);
+            }
+
+            @Override
+            public void onLongPress(View view, RegistrationPacketModel registrationPacketModel, int position) {
+                toggleActionBar(position);
+            }
+
+            @Override
+            public void onItemBtnPress(View view, RegistrationPacketModel registrationPacketModel, int position) {
+                syncAndUploadPacket(position);
+            }
+        });
+    }
+
+    /*
+       toggling action bar that will change the color and option
+     */
+    private void toggleActionBar(int position) {
+
+        if (actionMode == null) {
+            actionMode = startSupportActionMode(actionCallback);
+        }
+        toggleSelection(position);
+    }
+
+    /*
+       toggle selection of items and show the count of selected items on the action bar
+     */
+    private void toggleSelection(int position) {
+        adapter.toggleSelection(position);
+        int count = adapter.selectedItemCount();
+        if (count == 0) {
+            actionMode.finish();
+        } else {
+            actionMode.setTitle(String.valueOf(count));
+            actionMode.invalidate();
+        }
+    }
+
+    private void syncAndUploadSelectedPacket() {
+        if (!checkNetwork()) {
```

Comment:
Need to notify the user about the unavailability of the network.

Reply:
Toast msg is raised in "checkNetwork" method. line 291,296